src/main/java/io/github/thebusybiscuit/slimefun4/api/player/PlayerProfile.java

```diff
@@ -325,35 +325,52 @@ public final void markDirty() {
         return Optional.empty();
     }
 
-    // returns the amount of researches with at least 1 enabled item
-    private int nonEmptyResearches() {
-        return (int) Slimefun.getRegistry().getResearches()
-                .stream()
-                .filter(research -> research.getAffectedItems().stream().anyMatch(item -> item.getState() == ItemState.ENABLED))
-                .count();
+    private int countNonEmptyResearches(@Nonnull Collection<Research> researches) {
+        int count = 0;
+        for (Research research : researches) {
+            if (research.hasEnabledItems()) {
+                count++;
+            }
+        }
+        return count;
     }
 
+    /**
+     * This method gets the research title, as defined in {@code config.yml},
+     * of this {@link PlayerProfile} based on the fraction
+     * of unlocked {@link Research}es of this player.
+     *
+     * @return The research title of this {@link PlayerProfile}
+     */
     public @Nonnull String getTitle() {
         List<String> titles = Slimefun.getRegistry().getResearchRanks();
 
-        float fraction = (float) researches.size() / nonEmptyResearches();
+        int allResearches = countNonEmptyResearches(Slimefun.getRegistry().getResearches());
+        float fraction = (float) countNonEmptyResearches(researches) / allResearches;
         int index = (int) (fraction * (titles.size() - 1));
 
         return titles.get(index);
     }
 
+    /**
+     * This sends the statistics for the specified {@link CommandSender}
+     * to the {@link CommandSender}. This includes research title, research progress
+     * and total xp spent.
+     *
+     * @param sender The {@link CommandSender} for which to get the statistics and send them to.
+     */
     public void sendStats(@Nonnull CommandSender sender) {
-        Set<Research> unlockedResearches = getResearches();
-        int levels = unlockedResearches.stream().mapToInt(Research::getCost).sum();
-        int allResearches = nonEmptyResearches();
+        int unlockedResearches = countNonEmptyResearches(getResearches());
+        int levels = getResearches().stream().mapToInt(Research::getCost).sum();
```

Comment:
Any reason you left this as all researches regardless of enabled?

Reply:
well I figured that the player did spend the levels, so they should be counted.